packages/admin-api-client/src/admin-api-client.ts

```diff
@@ -0,0 +1,138 @@
+import {
+  createGraphQLClient,
+  CustomFetchAPI,
+  RequestParams as GQLClientRequestParams,
+} from "@shopify/graphql-client";
+
+import {
+  AdminAPIClient,
+  AdminAPIClientConfig,
+  AdminAPIClientRequestOptions,
+} from "./types";
+import {
+  DEFAULT_SDK_VARIANT,
+  DEFAULT_CLIENT_VERSION,
+  SDK_VARIANT_HEADER,
+  SDK_VARIANT_SOURCE_HEADER,
+  SDK_VERSION_HEADER,
+  DEFAULT_CONTENT_TYPE,
+  ACCESS_TOKEN_HEADER,
+} from "./constants";
+import {
+  getCurrentSupportedAPIVersions,
+  validateRequiredStoreDomain,
+  validateRequiredAccessToken,
+  validateApiVersion,
+  validateServerSideUsage,
+} from "./utilities";
+
+const httpRegEx = new RegExp("^(https?:)?//");
+
+export function createAdminAPIClient({
+  storeDomain,
+  apiVersion,
+  accessToken,
+  clientName,
+  customFetchAPI: clientFetchAPI,
+}: {
+  storeDomain: string;
+  apiVersion: string;
+  accessToken: string;
+  clientName?: string;
+  customFetchAPI?: CustomFetchAPI;
+}): AdminAPIClient {
+  const currentSupportedAPIVersions = getCurrentSupportedAPIVersions();
+
+  validateServerSideUsage();
+  validateRequiredStoreDomain(storeDomain);
+  validateApiVersion(currentSupportedAPIVersions, apiVersion);
+  validateRequiredAccessToken(accessToken);
+
+  const trimmedStoreDomain = storeDomain.trim();
+  const cleanedStoreDomain = httpRegEx.test(trimmedStoreDomain)
+    ? trimmedStoreDomain.substring(trimmedStoreDomain.indexOf("//") + 2)
+    : trimmedStoreDomain;
+
+  const generateApiUrl = (version?: string) => {
+    if (version) {
+      validateApiVersion(currentSupportedAPIVersions, version);
+    }
+
+    const urlApiVersion = (version ?? apiVersion).trim();
+
+    return `https://${cleanedStoreDomain}${
+      cleanedStoreDomain.endsWith("/") ? "" : "/"
+    }admin/api/${urlApiVersion}/graphql.json`;
+  };
+
+  const config: AdminAPIClientConfig = {
+    storeDomain: trimmedStoreDomain,
+    apiVersion,
+    accessToken,
+    headers: {
+      "Content-Type": DEFAULT_CONTENT_TYPE,
+      Accept: DEFAULT_CONTENT_TYPE,
+      [SDK_VARIANT_HEADER]: DEFAULT_SDK_VARIANT,
+      [SDK_VERSION_HEADER]: DEFAULT_CLIENT_VERSION,
+      ...(clientName ? { [SDK_VARIANT_SOURCE_HEADER]: clientName } : {}),
+      [ACCESS_TOKEN_HEADER]: accessToken,
+    },
+    apiUrl: generateApiUrl(),
+    clientName,
+  };
+
+  const graphqlClient = createGraphQLClient({
+    headers: config.headers,
+    url: config.apiUrl,
+    fetchAPI: clientFetchAPI,
+  });
+
+  const getHeaders: AdminAPIClient["getHeaders"] = (customHeaders) => {
```

Comment:
In this case, because all the methods in this client are mostly wrappers of other functions that are either defined locally in the client or requires client info, creating factories for them seems a bit much? But like you, I don't feel too strongly about this so I'm good with either way....